### PR TITLE
QA: Bumps Dependencies and Improves Static Analysis

### LINKED ISSUE #124: Removal of `container-interop/container-interop`
### Feature Request

|    Q        |   A
|------------ | ------
| New Feature | yes
| BC Break    | yes

#### Summary

This issue is created to keep track of the status of getting rid `container-interop` in the `laminas` components.

### PR DESCRIPTION
**Pull request by gsteel**
|    Q          |   A
|-------------- | ------
| QA            | yes

### Description

- Drops PHP 7.3
- Drops Service Manager v2
- Updates latest dev dependencies
- Adds template types for plugin manager
- Fixes newly surfaced type errors
- Drops Interop\Container
- Moves Service Manager from dev deps to required deps

### CODE REVIEW

File: src/ValidatorPluginManager.php [hunk 1]
```diff
@@ -239,7 +248,7 @@ class ValidatorPluginManager extends AbstractPluginManager
         \Zend\Validator\Bitwise::class      => Bitwise::class,
         \Zend\Validator\Callback::class     => Callback::class,
         \Zend\Validator\CreditCard::class   => CreditCard::class,
-        \Zend\Validator\Csrf::class         => Csrf::class,
+        \zend\validator\csrf::class         => Csrf::class,
```
Comment (Ocramius) [inline src/ValidatorPluginManager.php]: To be reverted

**Review by Ocramius — CHANGES_REQUESTED** [review]

File: src/ValidatorPluginManager.php (hunk 1 above)
Comment (gsteel) [reply src/ValidatorPluginManager.php]: Eagle eyes! I'm blaming `phpcbf` for that!

File: src/ValidatorPluginManager.php [hunk 2]
```diff
@@ -2,11 +2,13 @@
 
 namespace Laminas\Validator;
 
-use Interop\Container\ContainerInterface;
+use Interop\Container\ContainerInterface; // phpcs:ignore
```
Comment (boesing) [inline src/ValidatorPluginManager.php]: Should become PSR since we dropped interop with one of the servicemanager versions.

**Review by boesing — CHANGES_REQUESTED** [review]
Can you also replace the interop containers with PSR? that would be decent.

If this component does require interop as a composer dependency, this can be dropped as well 👌🏻

2 birds with one stone ❤️

**Comment by gsteel** [comment]
Is something wrong with the [laminas-ci action](https://github.com/laminas/laminas-continuous-integration-action/releases) 1.20 release? - Everything is passing locally

**Comment by Ocramius** [comment]
Yup: new got version broke everything 😩

We need to patch it to add https://stackoverflow.com/questions/71849415/i-cannot-add-the-parent-directory-to-safe-directory-in-git/71904131#71904131 to it.

**Comment by Ocramius** [comment]
FWIW, raised https://github.com/laminas/laminas-continuous-integration-action/pull/95, need a review :)

**Review by Ocramius — APPROVED** [review]

**Review by Ocramius — APPROVED** [review]
:ship: thanks @gsteel!

**Pull request merged**